### ISSUE TITLE
Unrecoverable error. No such object in /tmp/realm-java/realm/realm-library/src/main/cpp/io_realm_internal_OsList.cpp line 125

### ISSUE THREAD

**Issue by piotrros** [realm/realm-java#7506]
#### Goal
App wants to read some data from database.

#### Actual Results
I have an user, which have an exception:

```
Fatal Exception: io.realm.exceptions.RealmError
Unrecoverable error. No such object in /tmp/realm-java/realm/realm-library/src/main/cpp/io_realm_internal_OsList.cpp line 125
io.realm.internal.OsList.nativeGetRow (OsList.java)
io.realm.internal.OsList.getUncheckedRow (OsList.java)
io.realm.RealmModelListOperator.get (RealmModelListOperator.java)
io.realm.RealmList.get (RealmList.java)
```

It was reported in Firebase Crashlytics today.

#### Steps & Code to Reproduce

I cannot reproduce this issue on my own. This is happening only on one device. Function, which causes a crash looks like this:

```
public static String buildGpsString(RealmList<OfflineLocation> locations, int maxPoints) {
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append("["); //NON-NLS

        int size = locations.size();
        if(size > maxPoints) size = maxPoints;

        for (int i = 0; i < size; i++) {
            stringBuilder.append("["); //NON-NLS
            OfflineLocation lastLocation = locations.get(i);
            if(lastLocation == null) continue;
            stringBuilder.append(Utils.safeDoubleString(lastLocation.latitude));
            stringBuilder.append(","); //NON-NLS
            stringBuilder.append(Utils.safeDoubleString(lastLocation.longitude));
            stringBuilder.append(","); //NON-NLS
            stringBuilder.append("\""); //NON-NLS
            stringBuilder.append(lastLocation.formattedDate);
            stringBuilder.append("\""); //NON-NLS
            stringBuilder.append(","); //NON-NLS
            stringBuilder.append(Utils.safeDoubleString(lastLocation.altitude));
            stringBuilder.append(","); //NON-NLS
            stringBuilder.append(Utils.safeDoubleString(lastLocation.speed));

            stringBuilder.append(","); //NON-NLS
            stringBuilder.append(lastLocation.autopause);
            stringBuilder.append(","); //NON-NLS
            stringBuilder.append(Utils.safeDoubleString(lastLocation.accuracy));
            stringBuilder.append(","); //NON-NLS
            stringBuilder.append("\""); //NON-NLS
            stringBuilder.append(lastLocation.deviceDate);
            stringBuilder.append("\""); //NON-NLS

            stringBuilder.append("]"); //NON-NLS
            if (i + 1 < size) {
                stringBuilder.append(","); //NON-NLS
            }
        }
        stringBuilder.append("]"); //NON-NLS
        return stringBuilder.toString();
    }
```

It simply takes a list of locations and writes them into single string. Don't know exact line where it crashed.

#### Version of Realm and tooling

Realm version(s): ? 10.0.1

Realm Sync feature enabled: No

Android Studio version: 4.1.2

Android Build Tools version: 30.0.3

Gradle version: 4.1.2

Which Android version and device(s): myPhone Hammer Energy (Android 7.0)

More data about device:

```
Device
Brand:myPhone
Model:myPhone Hammer Energy
Orientation: Portrait
RAM free: 1.03 GB
Disk free: 1.08 GB
plat_android Operating System
Version:7.0
Orientation: Portrait
Rooted:No
```

**Comment by cmelchior** [comment]
Hi @piotrros 
Thank you for the report. We have had a few fixes between 10.0.1 and 10.6.0 that has been related to this area. Can you try to upgrade and see if that fixes the issue?

**Comment by pabvald3s** [comment]
@cmelchior Updating to realm-java version `1.14.0` fixed it for me. Thanks
